vendor/github.com/docker/docker/pkg/system/filesys_windows.go

```diff
@@ -103,13 +102,13 @@ func mkdirall(path string, applyACL bool, sddl string) error {
 // and Local System.
 func mkdirWithACL(name string, sddl string) error {
 	sa := windows.SecurityAttributes{Length: 0}
-	sd, err := winio.SddlToSecurityDescriptor(sddl)
+	sd, err := windows.SecurityDescriptorFromString(sddl)
```

Comment:
Thanks for explaining this to me. As mentioned I'm new to Go and did find it a bit strange that the vendor files are committed into the repo compared to how other languages work (e.g. the `dist` directory isn't normally committed when using npm).
It appears that the required change has been made in the [upstream library](https://github.com/moby/moby/blob/master/pkg/system/filesys_windows.go#L105), I'll work towards getting the module reference updated.

Reply:
It appears that the upstream library has the required change but an updated package has not yet been released.
I will close this PR until `github.com/docker/docker` is released with this change.

Reply:
> Thanks for explaining this to me. As mentioned I'm new to Go and did find it a bit strange that the vendor files are committed into the repo compared to how other languages work (e.g. the `dist` directory isn't normally committed when using npm).

By default they aren't. Only the versions and checksums of 3. party libraries are listed in the go.mod and go.sum files.

I prefer to create the offline copies of the libraries, to not have dependencies to 3. party services.
If github and the go module proxies became unavailable or the references are deleted, it would not be possible to build baur anymore.
It also speeds up CI tasks a bit, if the GOCACHE is not shared between all steps :-).


> It appears that the upstream library has the required change but an updated package has not yet been released.
I will close this PR until github.com/docker/docker is released with this change.

Could you explain to me why the change is a must have?
Without it, it's not possible to upload docker images at all?

Reply:
Sorry about the lack of response. I'm looking at supporting baur on Windows again now.
baur does not compile on Windows without this change
```
$ go build
# github.com/docker/docker/pkg/system
vendor\github.com\docker\docker\pkg\system\filesys_windows.go:112:24: cannot use uintptr(unsafe.Pointer(&sd[0])) (type uintptr) as type *"golang.org/x/sys/windows".SECURITY_DESCRIPTOR in assignment
```

Reply:
Ok, best might be switching to using the official docker client instead of fsouze/go-dockerclient (https://github.com/simplesurance/baur/issues/189).

All other issues that were fixed in this PR should already be in master. The filepath functions are used, Commands are run directly not via `/bin/sh` anymore.

Reply:
Having the Commands run directly has been great so far, thanks for making that change 💯 

Unfortunately the issue resides in the official docker client, not in fsouze/go-dockerclient.
This describes the breaking change to the golang.org/x/sys module https://github.com/golang/go/issues/34610
The fix (https://github.com/moby/moby/pull/40021) was merged into the official docker client master branch late last year  but has not been released yet. It appears it will be a part of v20 and has not been back-ported into any of the older versions.

There are 2 options that I can think of:
1. Wait until the official docker client has been released with the fix
2. Pin the golang.org/x/sys module to the version before the breaking change was made. Rancher have made this change (https://github.com/rancher/rke/pull/2204).

I have got baur compiling with option 2 and am working through the initial test failures on Windows